import/import_script.go

```diff
@@ -0,0 +1,897 @@
+			log.Fatalf("Invalid resource: %s is not accepted. The only accepted resources are %s", resource, acceptedResources)
+			return
+		}
+	}
+
+	log.Println("Resources to import: ", resources)
+
+	// set the API token
+	token := *tokenPtr
+	if token == "" {
+		token = os.Getenv("ASTRO_API_TOKEN")
+	}
+
+	if len(token) == 0 {
+		log.Fatal("API token not provided")
+		return
+	}
+
+	// set the host
+	var host string
+	if *hostPtr == "dev" {
+		host = "https://api.astronomer-dev.io"
+	} else if *hostPtr == "stage" {
+		host = "https://api.astronomer-stage.io"
+	} else {
+		host = *hostPtr
+	}
+
+	// set the organization ID
+	organizationId := *organizationIdPtr
+	if organizationId == "" {
+		log.Fatalf("Organization ID not provided")
+	}
+
+	log.Printf("Using organization ID: %s", organizationId)
+
+	// Check if Terraform is installed
+	_, err := exec.LookPath("terraform")
+	if err != nil {
+		log.Fatalf("Error: Terraform is not installed or not in PATH. Please install Terraform and make sure it's in your system PATH")
+	}
+
+	// connect to v1beta1 client
+	ctx := context.Background()
+	platformClient, err := platform.NewPlatformClient(host, token, "import")
+	if err != nil {
+		log.Fatalf("Failed to create platform client: %v", err)
+	}
+
+	iamClient, err := iam.NewIamClient(host, token, "import")
+	if err != nil {
+		log.Fatalf("Failed to create iam client: %v", err)
+		return
+	}
+
+	// set terraform provider configuration
+	var importString string
+	importString += fmt.Sprintf(`terraform {
+	required_providers {
+		astro = {
+			source = "astronomer/astro"
+		}
+	}
+}
+
+provider "astro" {
+	organization_id = "%s"
+	host = "%s"
+	token = "%s"
+}
+`, organizationId, host, token)
+
+	//	for each resource, we get the list of entities and generate the terraform import command
+
+	resourceHandlers := map[string]func(context.Context, *platform.ClientWithResponses, *iam.ClientWithResponses, string) (string, error){
+		"workspace":  handleWorkspaces,
+		"deployment": handleDeployments,
+		"cluster":    handleClusters,
+		"api_token":  handleApiTokens,
+		"team":       handleTeams,
+		"team_roles": handleTeamRoles,
+		"user_roles": handleUserRoles,
+	}
+
+	results := make(chan HandlerResult, len(resources))
+	var wg sync.WaitGroup
+
+	for _, resource := range resources {
+		wg.Add(1)
+
+		go func(resource string) {
+			defer wg.Done()
+			handler, exists := resourceHandlers[resource]
+			if !exists {
+				log.Printf("Resource not supported: %s", resource)
+				results <- HandlerResult{Resource: resource, Error: fmt.Errorf("resource not supported")}
+				return
+			}
+			result, err := handler(ctx, platformClient, iamClient, organizationId)
+			if err != nil {
+				log.Printf("Error handling resource %s: %v", resource, err)
+				results <- HandlerResult{Resource: resource, Error: err}
+			} else {
+				results <- HandlerResult{Resource: resource, ImportString: result}
+			}
+		}(resource)
+	}
+
+	go func() {
```

Comment:
curious why we need a goroutine here

Reply:
the block above parallelizes the import generation step for each resource in individual go routines

to ensure that all the go routines are finished, we need a separate goroutine to wait for all the goroutines to finish writing to the channel before closing the channel

Reply:
would it work the same without the goroutine?
to me, it seems like without go routine makes more sense so just 
```
wg.Wait()
close(results)
```

but maybe im missing something